src/edu/csus/ecs/pc2/core/InternalController.java

```diff
@@ -4988,22 +4988,50 @@ public IInternalContest getContest() {
     }
 
     @Override
-    public void submitRun(ClientId submitter, Problem problem, Language language, SerializedFile mainSubmissionFile, SerializedFile[] additionalFiles, long overrideTimeMS, long overrideRunId) {
+    public void submitRun(ClientId submitter, Problem problem, Language language, SerializedFile mainSubmissionFile, SerializedFile[] additionalFiles, long overrideTimeMS, long overrideRunId) throws Exception {
```

Comment:
As I mentioned in the other file(s), I think it would make more sense to have this throw `core.exception.SubmissionRejectedException`.

Reply:
I just did what eclipse suggested.  I guess if an other exception occurs the submissions was not accepted anyway.